Source/Initialization/WarpXInitData.cpp

```diff
@@ -306,3 +445,98 @@ WarpX::InitLevelDataFFT (int lev, Real time)
 }
 
 #endif
+
+void
+WarpX::InitializeExternalFieldsOnGridUsingParser (
+       MultiFab *mfx, MultiFab *mfy, MultiFab *mfz,
+       ParserWrapper *xfield_parser, ParserWrapper *yfield_parser,
+       ParserWrapper *zfield_parser, IntVect x_nodal_flag,
+       IntVect y_nodal_flag, IntVect z_nodal_flag,
+       const int lev)
+{
+
+    const auto dx_lev = geom[lev].CellSizeArray();
+    const RealBox& real_box = geom[lev].ProbDomain();
+    for ( MFIter mfi(*mfx, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+       const Box& tbx = mfi.tilebox(x_nodal_flag);
+       const Box& tby = mfi.tilebox(y_nodal_flag);
+       const Box& tbz = mfi.tilebox(z_nodal_flag);
+
+       auto const& mfxfab = mfx->array(mfi);
+       auto const& mfyfab = mfy->array(mfi);
+       auto const& mfzfab = mfz->array(mfi);
+
+       auto const& mfx_IndexType = (*mfx).ixType();
+       auto const& mfy_IndexType = (*mfy).ixType();
+       auto const& mfz_IndexType = (*mfz).ixType();
+
+       // Initialize IntVect based on the index type of multiFab
+       // 0 if cell-centered, 1 if node-centered.
+       IntVect mfx_type(AMREX_D_DECL(0,0,0));
+       IntVect mfy_type(AMREX_D_DECL(0,0,0));
+       IntVect mfz_type(AMREX_D_DECL(0,0,0));
+
+       for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+           mfx_type[idim] = mfx_IndexType.nodeCentered(idim);
+           mfy_type[idim] = mfy_IndexType.nodeCentered(idim);
+           mfz_type[idim] = mfz_IndexType.nodeCentered(idim);
+       }
```

Comment:
Could you explain what this is doing? In the end, why aren't we using `mfx_IndexType` directly?

Reply:
The IndexType is NodeCentered or CellCentered and not an integer. mfx_type converts the IndexType to an integer which is used to compute the exact x,y,z positions.